### PR TITLE
feat: Dataset naming migration

### PR DESCRIPTION
**Pull request by mayurnewase**
### SUMMARY

This pr adds migration required for setting names to datasets.
This is the first step for pr #14393 which intends to do the same.
We also need migration to set dataset_name to table_name for the existing datasets,which is better be done in #14393 .

### TEST PLAN

Test upgrade
Test Downgrade

### ADDITIONAL INFORMATION

- [x] Has associated issue:
- [ ] Changes UI
- [x] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [x] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/migrations/versions/d891f321997f_add_dataset_name_column_to_tables.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+def upgrade():
+    op.add_column(
+        "tables", sa.Column("dataset_name", sa.String(length=250), nullable=True)
+    )
+
+
+def downgrade():
+    op.drop_column("tables", "dataset_name")
```
Comment (mayurnewase) [inline superset/migrations/versions/d891f321997f_add_dataset_name_column_to_tables.py:40]: Sqlite does not support dropping column.
Don't know any way around.

**Comment by mayurnewase** [comment]
@etr2460

**Comment by mayurnewase** [comment]
@etr2460 Could you review please?

File: superset/migrations/versions/d891f321997f_add_dataset_name_column_to_tables.py (hunk 1 above)
Comment (dpgaspar) [reply superset/migrations/versions/d891f321997f_add_dataset_name_column_to_tables.py:40]: You can use this: https://alembic.sqlalchemy.org/en/latest/batch.html

**Pull request closed without merging**